cider/exec/nextgen/Transformer.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_EXEC_NEXTGEN_TRANSFORMER_H
+#define CIDER_EXEC_NEXTGEN_TRANSFORMER_H
+
+#include <memory>
+
+#include "exec/nextgen/OpPipeline.h"
+#include "exec/nextgen/Translator.h"
+
+namespace cider::exec::nextgen {
+
+/// \brief A transformer convert an OpPipeline to a translator
+class Transformer {
+ public:
+  static std::unique_ptr<Translator> toTranslator(const OpPipeline& pipeline) {
+    return std::make_unique<Translator>();
```

Comment:
Any other class/API needed here? form pipeline to Translator. I suggest to list some even it's internal method.